lib/upipe-modules/upipe_rtp_prepend.c

```diff
@@ -167,6 +167,33 @@ static void upipe_rtp_prepend_input(struct upipe *upipe, struct uref *uref,
     ubuf_block_unmap(header, 0);
     upipe_rtp_prepend->seqnum++;
 
+    const char *def;
+    ubase_assert(uref_flow_get_def(upipe_rtp_prepend->flow_def, &def));
+    if (!ubase_ncmp(def, "block.rtp.mp2.sound.") ||
+            !ubase_ncmp(def, "block.rtp.mp3.sound.")) {
+        /* alloc mpa header */
+        struct ubuf *mpa_header = ubuf_block_alloc(uref->ubuf->mgr, 4);
+        if (unlikely(!mpa_header)) {
+            upipe_throw_fatal(upipe, UBASE_ERR_ALLOC);
+            uref_free(uref);
+            ubuf_free(header);
+            return;
+        }
+
+        /* write mpa mpa_header */
+        ubuf_block_write(mpa_header, 0, &size, &buf);
+        memset(buf, 0, 4); // frag_offset = 0
```

Comment:
Here, it may be more efficient to allocate the ubuf just once, and for every subsequent packet call ubuf_dup() to get a shared pointer to the same area. That's what we do for TS padding for instance.